src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -207,40 +196,20 @@ export default class AirlineModel extends BaseModel {
 
     // TODO: the logic here can be simplified.
     /**
-     * Create a flight number/identifier
+     * Creates a flight number/identifier
      *
      * This method should only be called from the `AircraftController` so the controller
      * can guarantee unique `flightNumbers` across all `AirlineModels`.
      *
      * @for AirlineModel
      * @method generateFlightNumber
      * @return flightNumber {string}
-     */
+    */
     generateFlightNumber() {
-        let flightNumber = '';
-        let list = '0123456789';
-
-        // Start with a number other than zero
-        flightNumber += choose(list.substr(1));
-
-        if (!this.flightNumberGeneration.alphaNumeric) {
-            for (let i = 1; i < this.flightNumberGeneration.length; i++) {
-                flightNumber += choose(list);
-            }
-
-            return flightNumber;
-        }
-
-        // TODO: why `this.flightNumberGeneration.length - 3`?  enumerate the magic number.
-        for (let i = 0; i < this.flightNumberGeneration.length - 3; i++) {
-            flightNumber += choose(list);
-        }
-
-        list = 'abcdefghijklmnopqrstuvwxyz';
+        const flightNumber = flightNumberBuilder(this.flightNumberGeneration.callsignFormats);
 
-        // the end of an `N` style registration: `N322WT`
-        for (let i = 0; i < 2; i++) {
-            flightNumber += choose(list);
+        if (this.activeFlightNumbers.indexOf(flightNumber) !== -1) {
+            return this.generateFlightNumber();
```

Comment:
@n8rzz There seems to be two layers of recursion going on here. Here we start over if the `AirlineModel.activeFlightNumbers` contains the flight number, and we would also start over one step up in the call stack if `AirlineController.flightNumbers` contains the flight number. The latter is the one that matters, and come to think of it, there really is no reason we should even have a `AirlineModel.activeFlightNumbers` property. Most likely, it should be removed to avoid confusion, and then there is no need for the recursion at this line (212), as it is covered in the caller method.

Also important to note that the current behavior is that flight numbers are unique to all aircraft (which is to say we will not end up with EIN118 _AND_ BAW118, because `"118"`  is already "in use". Though not realistic, this enables an important feature of being able to issue commands by typing partial callsigns rather than clicking around.

Reply:
done